blocks/table/src/components/table/table-with-query.tsx

```diff
@@ -72,26 +80,75 @@ export const TableWithQuery = ({
         throw new Error(res.errors?.[0]?.message ?? "Unknown error");
       }
 
-      const roots = getRoots(res.data.results);
-      setLoading(false);
+      /** @todo: fix the return type */
+      const fetchedSubgraph = res.data as unknown as Subgraph<EntityRootType>;
```

Comment:
Can we fix it in `knowledge-shim.ts`? The problem is that the patch there has been defined to have a type which is not consistent with the type defined in the BP. 

It should be
```ts
export type QueryEntitiesMessageCallback = MessageCallback<
  QueryEntitiesRequest,
  null,
  MessageReturn<{
    results: Subgraph<EntityRootType>;
    operation: QueryOperationInput;
  }>,
  ReadOrModifyResourceError
>;
```

then we can update `useBlockProtocolQueryEntities` to return the new shape, and update the couple other places we use the hook to take `data.results` instead of just `data`. Saves this tripping us up in the future

Reply:
Ah thank you for explaining why this was happening, have addressed in https://github.com/hashintel/hash/pull/3753/commits/f66360e9ce3d002be737693c741e10aa41b2087c